server/odc-service/src/main/java/com/oceanbase/odc/service/flow/FlowInstanceService.java

```diff
@@ -970,6 +1076,69 @@ private FlowInstanceConfigurer buildConfigurer(
         return configurers.get(0);
     }
 
+    private FlowInstanceConfigurer buildConfigurerForMultipleDatabase(
+            @NonNull ApprovalFlowConfig approvalFlowConfig,
+            @NonNull FlowInstance flowInstance,
+            @NonNull TaskType taskType,
+            @NonNull Long taskId,
+            @NonNull TaskParameters parameters,
+            @NonNull CreateFlowInstanceReq flowInstanceReq) {
+        List<ApprovalNodeConfig> nodeConfigs = approvalFlowConfig.getNodes();
+        Verify.verify(!nodeConfigs.isEmpty(), "Approval Nodes size can not be equal to zero");
+        List<FlowInstanceConfigurer> configurers = new LinkedList<>();
+        for (int nodeSequence = 0; nodeSequence < nodeConfigs.size(); nodeSequence++) {
+            FlowInstanceConfigurer configurer;
+            ApprovalNodeConfig nodeConfig = nodeConfigs.get(nodeSequence);
+            Long resourceRoleId = nodeConfig.getResourceRoleId();
+            FlowApprovalInstance approvalInstance = flowFactory.generateFlowApprovalInstance(flowInstance.getId(),
+                    false, false,
+                    nodeConfig.getAutoApproval(), approvalFlowConfig.getApprovalExpirationIntervalSeconds(),
+                    nodeConfig.getExternalApprovalId());
+            if (Objects.nonNull(resourceRoleId)) {
+                approvalInstance.setCandidate(StringUtils.join(flowInstanceReq.getProjectId(), ":", resourceRoleId));
+            }
+            FlowGatewayInstance approvalGatewayInstance =
+                    flowFactory.generateFlowGatewayInstance(flowInstance.getId(), false, true);
+            configurer = flowInstance.newFlowInstanceConfigurer(approvalInstance);
+            configurer = configurer.next(approvalGatewayInstance).route(String.format("${!%s}",
+                    FlowApprovalInstance.APPROVAL_VARIABLE_NAME), flowInstance.endFlowInstance());
+            if (nodeSequence == nodeConfigs.size() - 1) {
+                FlowInstanceConfigurer taskConfigurer = null;
+                int orders = ((MultipleDatabaseChangeParameters) flowInstanceReq.getParameters())
```

Comment:
size value may not match the concept 'orders'

Reply:
This is the last approval node generated for a given risk level, independent of the concept order